EU4toV2/Source/V2World/Country/Country.cpp

```diff
@@ -0,0 +1,1033 @@
+	if (!match)
+	{
+		LOG(LogLevel::Warning) << "No religion mapping defined for " << srcReligion << " (" << srcCountry->getTag() << " -> " << tag << ')';
+	}
+	else
+	{
+		details.religion = *match;
+	}
+}
+
+void V2::Country::setPrimaryAndAcceptedCultures(std::shared_ptr<EU4::Country> srcCountry, const mappers::CultureMapper& cultureMapper, const EU4::Regions& eu4Regions)
+{
+	int oldCapital = srcCountry->getCapital();
+
+	// primary culture
+	std::string primCulture = srcCountry->getPrimaryCulture();
+	std::optional<std::string> matched = cultureMapper.cultureMatch(eu4Regions, primCulture, details.religion, oldCapital, srcCountry->getTag());
+	if (!matched)
+	{
+		LOG(LogLevel::Warning) << "No culture mapping defined for " << primCulture << " (" << srcCountry->getTag() << " -> " << tag << ')';
+	}
+	else
+	{
+		details.primaryCulture = *matched;
+	}
+
+	//accepted cultures
+	std::vector<std::string> srcAceptedCultures = srcCountry->getAcceptedCultures();
+	auto culturalUnion = srcCountry->getCulturalUnion();
+	if (culturalUnion)
+	{
+		for (auto unionCulture : culturalUnion->getCultures())
+		{
+			srcAceptedCultures.push_back(unionCulture.first);
+		}
+	}
+	for (auto srcCulture : srcAceptedCultures)
+	{
+		std::optional<std::string> dstCulture;
+		dstCulture = cultureMapper.cultureMatch(
+			eu4Regions,
+			srcCulture,
+			 details.religion,
+			oldCapital,
+			srcCountry->getTag()
+		);
+		if (dstCulture)
+		{
+			if (details.primaryCulture != *dstCulture) details.acceptedCultures.insert(*dstCulture);
+		}
+		else
+		{
+			LOG(LogLevel::Warning) << "No culture mapping defined for " << srcCulture << " (" << srcCountry->getTag() << " -> " << tag << ')';
+		}
+	}
+}
+
+std::tuple<double, double, double> V2::Country::getNationalValueScores()
+{
+	double orderScore = 0.0;
+	double libertyScore = 0.0;
+	double equalityScore = 0.0;
+
+	if (srcCountry)
+	{
+		orderScore += srcCountry->getOrderInvestment() - 5.0;
+		libertyScore += srcCountry->getLibertyInvestment() - 5.0;
+		equalityScore += srcCountry->getEqualityInvestment() - 5.0;
+	}
+
+	return std::make_tuple(libertyScore, equalityScore, orderScore);
+}
+
+void V2::Country::addRelation(Relation& newRelation)
+{
+	relations.insert(std::make_pair(newRelation.getTarget(), newRelation));
+}
+
+V2::Relation& V2::Country::getRelation(const std::string& target)
+{
+	const auto& relation = relations.find(target);
+	if (relation != relations.end()) return relation->second;
+	Relation newRelation(target);
+	relations.insert(std::make_pair(target, newRelation));
+	const auto& newRelRef = relations.find(target);
+	return newRelRef->second;
+}
+
+void V2::Country::absorbVassal(std::shared_ptr<Country> vassal)
+{
+	// change province ownership and add owner cores if needed
+	for (auto& province : vassal->getProvinces())
+	{
+		province.second->setOwner(tag);
+		province.second->setController(tag);
+		province.second->addCore(tag);
+	}
+	vassal->provinces.clear();
+
```

Comment:
Just to be clear, it's intentional to no longer take accepted cultures from vassals, right?

Reply:
Very much so. absorb vassal is a misnomer, should be absorbColony. Renamed it. 
England cares not what cultures Canada accepted, and even if it did, do they account for 15% of development? Not a clue, we left development behind with EU4 - this is the age of native assimilation.